### PR TITLE
Bump samlify, remove unused server deps, remove gql2ts (20%)

### PR DESCRIPTION
**Pull request by mattkrick**
**This is low priority.**
I had an extra hour this morning & wanted to do some cleaning up. 

What it does:
- Bumps samlify, which fixes a vuln in xmldom
- Removes unused dependencies from server/package.json. This helps with the privately managed instances. Fewer deps = fewer vulns 
- Removes gql2ts. We haven't used this in a couple years, but we were still building gql types every time we started up the dev server. Faster dev server startups!
- Refactors s3.js into the S3Manager. s3.js wasn't in typescript, 80% unused, and started an s3 instance when the server booted up.

TEST
- [ ] can update your profile picture on staging (to test that the S3Manager still works)

### CODE REVIEW

File: packages/server/utils/s3.js [hunk 1]
```diff
@@ -1,113 +0,0 @@
-import aws from 'aws-sdk'
-import promisify from 'es6-promisify'
-import mime from 'mime-types'
-import protocolRelativeUrl from './protocolRelativeUrl'
-
-/*
- * Initializing AWS S3 implicitly uses environment variables
- * AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
- */
-const s3 =
```
Comment (mattkrick) [inline packages/server/utils/s3.js:10]: creating things on server startup isn't great, now it gets created when the S3 manager is created

File: packages/server/utils/updateGQLSchema.ts [hunk 2]
```diff
@@ -16,27 +9,10 @@ const write = promisify(fs.writeFile)
...
-  interfaceBuilder: (name, body) => DEFAULT_EXPORT_FUNCTION(DEFAULT_INTERFACE_BUILDER(name, body)),
-  typeBuilder: (name, body) => DEFAULT_EXPORT_FUNCTION(DEFAULT_TYPE_BUILDER(name, body)),
-  enumTypeBuilder: (name, values) =>
-    DEFAULT_EXPORT_FUNCTION(DEFAULT_ENUM_TYPE_BUILDER(name, values))
-}
 
 const updateGQLSchema = async () => {
   const nextSchema = printSchema(schema)
-  const gqlTypes = generateNamespace('GQL', nextSchema, {}, typesOverrides)
-  await Promise.all([write(schemaPath, nextSchema), write(typesPath, gqlTypes)])
```
Comment (mattkrick) [inline packages/server/utils/updateGQLSchema.ts:39]: we have a large schema. this was turning the entire schema into typedefs. that was expensive & we didn't even use it

**Comment by elielytz** [comment]
## Test results
Passed: This works like a charm!! 

### Enhancement suggestion
Meanwhile, after the user updates the new profile picture, s/he has to click outside the icon setting pop window to close it on both mobile and desktop.

Screenshot from 2022-01-28 12-37-19

One of the following options can be added to enhance the experience;

- Close sign icon
- Done button
-  Notification prompts user to "click outside the pop-up window to close it"

**Comment by mattkrick** [comment]
thanks @elielytz! a close button is a really good idea, we've got a few of these old modals that don't have them.
if you like, could you crate an issue for that? we can handle that in the next sprint or 2

**Comment by elielytz** [comment]
Sure, let me create this enhancement issue.

**Pull request merged**